design/hostclaim-multitenancy.md

```diff
@@ -0,0 +1,582 @@
+  labels. The namespace of the HostClaim must have a label belonging to the
+  list.
+
+To successfully match, the namespace must fulfill the constraints of all the
+fields that are defined. An empty HostDeployPolicy authorizes all namespaces.
+A HostClaim is authorized to bind a BareMetalHost if and only if the namespace
+of the HostClaim is successfully filtered by at least one HostDeployPolicy in
+the namespace of the BareMetalHost. In that case, the binding between the
+HostClaim and the BareMetalHost is authorized.
+
+A HostDeployPolicy specification also contains a field ``metadataCopy``. It defines
+rules for which annotations are copied (``annotationsMatch``) and which
+labels are copied (``labelsMatch``). Both fields contain regular expressions.
+
+For example,
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HostDeployPolicy
+metadata:
+  name: policy
+  namespace: infra
+spec:
+  hostclaimNamespaces:
+    namespaces:
+    - cluster1
+    - cluster2
+  metadataCopy:
+    annotationsMatch: 'mydomain/.*'
+    labelsMatch: 'selectors/.*'
+```
+
+will make the BareMetalHosts in namespace ``infra`` to HostClaims in namespaces
+``cluster1`` and ``cluster2``. All labels in domain ``selectors`` will be
+available for selection as annotations in ``mydomain`` for constructing
+cloud-init ``metadata``.
+
+#### Multi-tenancy
+
+As an infrastructure administrator I would like to host several isolated
+clusters.
+
+All the servers in the data-center are registered as BareMetalHost in one or
+several namespaces under the control of the infrastructure manager. Namespaces
+are created for each tenant of the infrastructure. They create
+standard cluster definitions in those namespaces.
+
+When the cluster is started, a HostClaim is created for each Metal3Machine
+associated to the cluster. The ``hostSelector`` is inherited from the
+Metal3Machine. As in the original workflow, it is used to choose the BareMetalHost
+associated with the cluster, but the associated BareMetalHost is not in the same
+namespace as the HostClaim. The exact definition of the BareMetalHost remains
+hidden from the cluster user, only a small parts of its status is copied
+back to the HostClaim resource (actual power state). The status of the HostClaim
+contains a reference to the BareMetalHost and so to the HardwareData that is
+readable by the end user.
+
+With the help of HardwareData, the data template controller has enough details to
+compute the different secrets (userData, metaData and networkData) associated to
+the Metal3Machine. Those secrets are linked to the HostClaim and, ultimately, to
+the BareMetalHost.
+
+When the cluster is modified, new Machine and Metal3Machine resources replace
+the previous ones. The HostClaims follow the life-cycle of the Metal3Machines
+and are destroyed with them. The BareMetalHosts are recycled and are bound to
+new HostClaims, potentially belonging to other clusters.
+
+#### Compute Resources in Another Cluster
```

Comment:
I assume this part is now covered by [Lennart's proposal?](https://github.com/metal3-io/metal3-docs/pull/429), so maybe link to it and avoid duplicate information here?

Reply:
An explicit pointer on the document yes, but it must be reimplemented at the level of HostClaims so I think it cannot be completely removed and it must be clear that the Metal3Machine creates a HostClaim on a remote cluster, not that the HostClaim creates a remote BMH.